packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/cloudwatch_logger_plugin.dart

```diff
@@ -200,43 +173,49 @@ class CloudWatchLoggerPlugin extends AWSLoggerPlugin
 
   Future<PutLogEventsResponse> _sendToCloudWatch(
     List<InputLogEvent> logEvents,
-    String logStreamName,
   ) async {
+    final logStreamName = await _logStreamProvider.logStream;
     final request = PutLogEventsRequest(
       logGroupName: _pluginConfig.logGroupName,
       logStreamName: logStreamName,
       logEvents: logEvents,
     );
-    return _client.putLogEvents(request).result;
+    try {
+      return _client.putLogEvents(request).result;
+    } on ResourceNotFoundException {
+      await _logStreamProvider.createLogStream(logStreamName);
+      return _client.putLogEvents(request).result;
+    } on Exception {
+      rethrow;
+    }
   }
 
   Stream<_LogBatch> _getLogBatchesToSync() async* {
     final queuedLogs = (await _logStore.getAll()).toList();
-    if (queuedLogs.isEmpty) {
-      yield ([], []);
-    }
-    final logEvents = <InputLogEvent>[];
-    final logQueues = <QueuedItem>[];
-    var totalByteSize = 0;
-
-    for (final currentLog in queuedLogs) {
-      final currentLogEvent = currentLog.toInputLogEvent();
-      final size = currentLogEvent.message.length + _baseBufferSize;
-      if (totalByteSize + size >= _maxLogEventsBatchSize ||
-          logEvents.length >= _maxNumberOfLogEventsInBatch ||
-          (logEvents.length > 1 &&
-              currentLogEvent.timestamp - logEvents.first.timestamp >=
-                  _maxLogEventsTimeSpanInBatch)) {
-        yield (logQueues, logEvents);
-        totalByteSize = 0;
-        logEvents.clear();
-        logQueues.clear();
+    if (queuedLogs.isNotEmpty) {
```

Comment:
Prefer exit early to improve readability and decrease average indent level

```suggestion
    if (queuedLogs.isEmpty) {
      return;
    }
```